extensions/blocks/social-previews/modal.js

```diff
@@ -0,0 +1,78 @@
+/**
+ * Social Previews modal component.
+ *
+ * Shows individual previews in modal window.
+ */
+
+/**
+ * External dependencies
+ */
+import { __ } from '@wordpress/i18n';
+import { Modal, TabPanel } from '@wordpress/components';
+import { withSelect } from '@wordpress/data';
+
+/**
+ * Internal dependencies
+ */
+import SocialPreviewsUpgrade from './upgrade';
+import { AVAILABLE_SERVICES } from './constants';
+import { getMediaSourceUrl } from './utils';
+
+const SocialPreviewsModal = function SocialPreviewsModal( {
+	onClose,
+	showUpgradeNudge,
+	image,
+	title,
+	description,
+	url,
+	author,
+} ) {
+	return (
+		<Modal
+			onRequestClose={ onClose }
+			title={ __( 'Social Previews', 'jetpack' ) }
+			className="jetpack-social-previews__modal"
+		>
+			{ showUpgradeNudge ? (
+				<SocialPreviewsUpgrade />
+			) : (
+				<TabPanel className="jetpack-social-previews__tabs" tabs={ AVAILABLE_SERVICES }>
+					{ tab => (
+						<div>
+							<tab.preview
+								title={ title }
+								description={ description }
+								url={ url }
+								author={ author }
+								image={ image }
```

Comment:
This is where we might need some defaults or a fallback experience if the post is empty.

Reply:
Yeah, but I wonder if we should actually shift this responsibility to the social-previews package. I'd love if all preview components took the exact same props and decide on their own what to render, what fallback they used, if they show no/small/big image etc… Right now we'll have to maintain many of those aspects separately in Calypso preview and Social Previews

Reply:
I would say that is a good idea. We started trying to standardize things in the initial port, but if we're noticing more stuff then now is a good time to create those followup Issues for next Iteration.

Reply:
Let's write followup tasks for this on Monday AM!

Reply:
Task https://github.com/Automattic/wp-calypso/issues/44798